python/paddle/fluid/regularizer.py

```diff
@@ -231,6 +263,95 @@ def __str__(self):
         return "L1Decay, regularization_coeff=%f" % self._regularization_coeff
 
 
+class WeightDecayBase(object):
+    def __str__(self):
+        raise NotImplementedError()
+
+    def __call__(self, params_and_grads):
+        raise NotImplementedError()
+
+    def apply(self):
+        raise NotImplementedError()
+
+
+class WeightDecay(WeightDecayBase):
```

Comment:
I don't think this class need a base class.

It is no need to have an interface when there is only one implementation if we are not using JAVA.

Reply:
BTW, I think maybe we should only use one scale operator instead of element_mul + assign considering memory consumption.